Use application/sd-cwt and define +sd-cwt (#106)
ietf-wg-spice/draft-ietf-spice-sd-cwt: draft-ietf-spice-sd-cwt.md

    title: "t-Closeness: Privacy Beyond k-Anonymity and l-Diversity"
    date: 2007-06-04

>   IANA.COSE:
>     target: https://www.iana.org/assignments/cose
>     title: CBOR Object Signing and Encryption (COSE)
>     author:
>     - org: IANA
> 
>   IANA.CBOR.Simple:
>     target: https://www.iana.org/assignments/cbor-simple-values
>     title: CBOR Simple Values
>     author:
>     - org: IANA
> 
>   IANA.CBOR.Tags:
>     target: https://www.iana.org/assignments/cbor-tags
>     title: CBOR Tags
>     author:
>     - org: IANA
> 
>   IANA.CWT:
>     target: https://www.iana.org/assignments/cwt
>     title: CBOR Web Token (CWT) Claims
>     author:
>     - org: IANA
> 
>   IANA.MediaTypes:
>     target: https://www.iana.org/assignments/media-types
>     title: Media Types
>     author:
>     - org: IANA
> 
>   IANA.StructuredSuffix:
>     target: https://www.iana.org/assignments/media-type-structured-suffix
>     title: Structured Syntax Suffixes
>     author:
>     - org: IANA
> 
>   IANA.Core.Parameters:
>     target: https://www.iana.org/assignments/core-parameters
>     title: Constrained RESTful Environments (CoRE) Parameters
>     author:
>     - org: IANA

Reviewer: I really don't link the way this renders as a reference instead of an external link. The way they are in the draft now isn't great, but this is not an improvement IMO. I'll propose something below. Also, in general I think we should include the link fragments in the URLs to go directly to the specific registries.

```suggestion
```

Author: Inline links don't work at all in the .txt output of the RFC.  This is the style that the RFC Editor uses so that the IANA references work in all the output formats.

If we want link fragments (which the RFC Editor often deletes), we would need separate references for each registry, rather than for the registry page.

Reviewer: >Inline links don't work at all in the .txt output of the RFC.
That's not the case.

Here is a sample of unmodified .txt generated from the markdown style I proposed:

"16.5.  Media Types

   This section requests the registration of new media types in the IANA
   "Media Types" registry (https://www.iana.org/assignments/media-types/
   media-types.xhtml)."

>If we want link fragments (which the RFC Editor often deletes), we would need separate references for each registry, rather than for the registry page.

We only need one extra reference in this document fortunately. I am willing to push for link fragments with the RFC editor because they have important meaning here.

# SD-CWT Definition {#sd-cwt-definition}

SD-CWT is modeled after SD-JWT, with adjustments to align with conventions in CBOR, COSE, and CWT.
An SD-CWT MUST include the protected header parameter `typ` {{!RFC9596}} with a value declaring that the object is an SD-CWT.
This value MAY be the string content type value `application/sd-cwt`,

> the uint CoAP content-format value TBD11,

Reviewer: ```suggestion
the uint Constrained Application Protocol (CoAP) {{?RFC7252}} content-format value TBD11,
```

## COSE Header Parameters

> IANA is requested to add the following entries to the IANA "COSE Header Parameters" registry {{IANA.COSE}}:

Reviewer: ```suggestion
IANA is requested to add the following entries to the [IANA "COSE Header Parameters" registry](https://www.iana.org/assignments/cose/cose.xhtml#header-parameters):
```

Author: As I said earlier, this doesn't work at all in the .txt output for the RFC.

## CBOR Simple Values {#simple59}

IANA is requested to add the following entry to the IANA "CBOR Simple Values" registry {{IANA.CBOR.Simple}}:

* Value: TBD4 (requested assignment 59)
* Semantics: This value as a map key indicates that the claim value is an array of redacted claim keys at the same level as the map key.
* Specification Document(s): {{blinded-claims}} of this specification

## CBOR Tags

> IANA is requested to add the following entries to the IANA "CBOR Tags" registry {{IANA.CBOR.Tags}}:

Author: Ditto

## CBOR Web Token (CWT) Claims

> IANA is requested to add the following entry to the IANA "CBOR Web Token (CWT) Claims" registry {{IANA.CWT}}:

Reviewer: ```suggestion
IANA is requested to add the following entry to the [IANA "CWT Claims" registry](https://www.iana.org/assignments/cwt/cwt.xhtml#claims-registry):
```

Author: Ditto

## Media Types

> IANA is requested to add the following entries to the IANA "Media Types" registry {{IANA.MediaTypes}}:

Reviewer: ```suggestion
IANA is requested to add the following entries to the IANA "Media Types" registry (https://www.iana.org/assignments/media-types/media-types.xhtml#application):
```

- provide test vectors **TODO**
- add AEAD and COSE encrypted disclosures
- Applied clarifications and corrections suggested by Mike Jones.
- Do not update CWT {{!RFC8392}}.

> - Use `application/sd-cwt` media type and define `+sd-cwt` structured suffix.

Reviewer: @selfissued as an editor style note, I usually do not update the change log in each PR when there are multiple PRs in flight, because this inevitably creates merge conflicts. I usually do a separate PR shortly before submitting an I-D, unless I expect only a single PR with changes worthy of the change log at a time. YMMV.
